include/nanoeigenpy/decompositions/sparse/SimplicialCholesky.hpp

```diff
@@ -0,0 +1,88 @@
+/// Copyright 2025 INRIA
+#pragma once
+
+#include "nanoeigenpy/decompositions/base.hpp"
+#include "nanoeigenpy/decompositions/sparse/SparseSolverBase.hpp"
+
+#include <Eigen/SparseCholesky>
+
+namespace nanoeigenpy {
+
+struct SimplicialCholeskyVisitor : nb::def_visitor<SimplicialCholeskyVisitor> {
+  template <typename SimplicialDerived, typename... Ts>
+  void execute(nb::class_<SimplicialDerived, Ts...> &cl) {
+    using Solver = SimplicialDerived;
+    using MatrixType = typename SimplicialDerived::MatrixType;
+    // using Scalar = typename MatrixType::Scalar;
+    using RealScalar = typename MatrixType::RealScalar;
+    // using DenseVectorXs = Eigen::Matrix<Scalar, Eigen::Dynamic, 1,
+    // MatrixType::Options>; using DenseMatrixXs = Eigen::Matrix<Scalar,
+    // Eigen::Dynamic, Eigen::Dynamic,
+    //                       MatrixType::Options>;
+
+    cl.def("analyzePattern", &Solver::analyzePattern,
+           "Performs a symbolic decomposition on the sparcity of matrix.\n"
+           "This function is particularly useful when solving for several "
+           "problems having the same structure.")
+
+        .def(EigenBaseVisitor())
+        .def(SparseSolverBaseVisitor())
+
+        .def(
+            "matrixL",
+            [](Solver const &self) -> MatrixType { return self.matrixL(); },
+            "Returns the lower triangular matrix L.")
+        .def(
+            "matrixU",
+            [](Solver const &self) -> MatrixType { return self.matrixU(); },
+            "Returns the upper triangular matrix U.")
+
+        .def(
+            "compute",
+            [](Solver &self, MatrixType const &matrix) {
```

Comment:
You should return a reference to `Solver`, here. Otherwise C++ infers that you want to return by value the type returned from `self.compute(matrix)` which is `Solver`.

In this case this triggers a compilation error because the copy constructor is deleted.